api/src/routes/challenge.ts

```diff
@@ -367,5 +368,63 @@ export const challengeRoutes: FastifyPluginCallbackTypebox = (
     }
   );
 
+  fastify.post(
+    '/save-challenge',
+    {
+      schema: schemas.saveChallenge,
+      errorHandler(error, request, reply) {
+        if (error.validation) {
+          void reply.code(400);
+          return formatProjectCompletedValidation(error.validation);
+        } else {
+          fastify.errorHandler(error, request, reply);
+        }
+      }
+    },
+    async (req, reply) => {
+      try {
+        const { files, id: challengeId } = req.body;
+        const user = await fastify.prisma.user.findUniqueOrThrow({
+          where: { id: req.session.user.id }
+        });
+        const challenge = {
+          id: challengeId,
+          files
+        };
+
+        if (!multifileCertProjectIds.includes(challengeId)) {
+          void reply.code(403);
+          return {
+            type: 'error',
+            message: 'That challenge type is not savable.'
+          } as const;
```

Comment:
```suggestion
          return 'That challenge type is not savable.';
```
It's silly and we will be changing this ASAP, but the api-server really does just return a string.